core/src/main/java/org/elasticsearch/index/fielddata/ScriptDocValues.java

```diff
@@ -253,124 +254,41 @@ public int size() {
             return values.count();
         }
 
-        public double factorDistance(double lat, double lon) {
-            GeoPoint point = getValue();
-            return GeoDistance.FACTOR.calculate(point.lat(), point.lon(), lat, lon, DistanceUnit.DEFAULT);
-        }
-
-        public double factorDistanceWithDefault(double lat, double lon, double defaultValue) {
-            if (isEmpty()) {
-                return defaultValue;
-            }
-            GeoPoint point = getValue();
-            return GeoDistance.FACTOR.calculate(point.lat(), point.lon(), lat, lon, DistanceUnit.DEFAULT);
-        }
-
-        public double factorDistance02(double lat, double lon) {
-            GeoPoint point = getValue();
-            return GeoDistance.FACTOR.calculate(point.lat(), point.lon(), lat, lon, DistanceUnit.DEFAULT) + 1;
-        }
-
-        public double factorDistance13(double lat, double lon) {
-            GeoPoint point = getValue();
-            return GeoDistance.FACTOR.calculate(point.lat(), point.lon(), lat, lon, DistanceUnit.DEFAULT) + 2;
-        }
-
         public double arcDistance(double lat, double lon) {
```

Comment:
> I also think not having instance methods will allow us to play more with the underlying field access so we dont need an intermediate object, GeoPoint).

+1 I'm working toward this over the course of a few "cleanup" PRs. But at the moment even doc['loc'] allocates a `GeoPoint` so cutting over to the static doesn't fix this issue. 

> One obvious way to calculate distance

Definitely agree. IMHO the .arcDistance is more obvious than the verbose static, and the `withDefault` methods also make handling missing values more obvious than having to put `if (doc['loc'] == null)` in the script? Maybe we also tackle this one in another PR?